sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -692,6 +693,45 @@ public virtual async Task StopProcessingAsync(CancellationToken cancellationToke
                         tasks.RemoveAll(t => t.IsCompleted);
                     }
                 }
+
+                // If the main loop is aborting due to the connection being canceled, then
+                // force the processor to stop.
+                if ((!cancellationToken.IsCancellationRequested) && (Connection.IsClosed))
+                {
+                    Logger.ProcessorClientClosedException(Identifier);
+
+                    // Because this is a highly unusual situation
+                    // and goes against the goal of resiliency for the processor, surface a
+                    // unique exception with explicit message, passing the standard
+                    // ObjectDisposedException as the inner exception.
+                    var fatalException = new ServiceBusException(
+                        false,
+                        Resources.DisposedConnectionMessageProcessorMustStop,
+                        null,
+                        ServiceBusFailureReason.ServiceCommunicationProblem,
+                        new ObjectDisposedException(nameof(ServiceBusConnection)));
+
+                    await InvokeErrorHandler(fatalException, ServiceBusErrorSource.Receive).ConfigureAwait(false);
+
+                    // This call will deadlock if awaited; StopProcessingAsync awaits
+                    // completion of this task.  An exception while stopping will already be logged,
+                    // attach a continuation that will invoke the customer error handler as well.
+                    _ = StopProcessingAsync()
```

Comment:
This is where things get a bit odd.  My goal is to trigger `StopProcessingAsync` as a fire-and-forget task, but attach a continuation that gets triggered if it throws, allowing us to signal the customer exception handler.   I would definitely appreciate a second opinion on the `ContinueWith` options.